test/governance/InterchainMultisig.js

```diff
@@ -0,0 +1,256 @@
+const chai = require('chai');
+const { ethers, network } = require('hardhat');
+const { sortBy } = require('lodash');
+const { getAddresses, encodeInterchainCallsBatch, getWeightedSignaturesProof } = require('../utils');
+const {
+    utils: { Interface },
+} = ethers;
+const { expect } = chai;
+
+describe('InterchainMultisig', () => {
+    const threshold = 2;
+
+    let wallets;
+    let owner;
+    let signers;
+    let newSigners = [];
+
+    let interchainMultisigFactory;
+    let interchainMultisig;
+    let targetFactory;
+    let targetContract;
+    let calldata;
+
+    before(async () => {
+        wallets = await ethers.getSigners();
+
+        owner = wallets[0];
+        signers = sortBy(wallets.slice(1, 3), (wallet) => wallet.address.toLowerCase());
+        newSigners = sortBy(wallets.slice(0, 2), (wallet) => wallet.address.toLowerCase());
+
+        interchainMultisigFactory = await ethers.getContractFactory('TestInterchainMultisig', owner);
+        targetFactory = await ethers.getContractFactory('Target', owner);
+
+        const targetInterface = new Interface(['function callTarget() external']);
+        calldata = targetInterface.encodeFunctionData('callTarget');
+    });
+
+    beforeEach(async () => {
+        // new multisig and target contracts for each test
+        interchainMultisig = await interchainMultisigFactory.deploy('Ethereum', [
```

Comment:
many tests can reuse the same multisig contract, group those under one describe block so you can deploy just once. this speeds up live network tests significantly